ext/mysql2/client.c

```diff
@@ -1465,3 +1467,30 @@ void init_mysql2_client() {
       LONG2NUM(CLIENT_BASIC_FLAGS));
 #endif
 }
+
+#define flag_to_bool(f) ((client->server_status & f) ? Qtrue : Qfalse)
+
+void rb_mysql_set_server_query_flags(MYSQL *client, VALUE result) {
+  VALUE server_flags = rb_hash_new();
+
+#ifdef SERVER_QUERY_NO_GOOD_INDEX_USED
+  rb_hash_aset(server_flags, ID2SYM(rb_intern("no_good_index_used")), flag_to_bool(SERVER_QUERY_NO_GOOD_INDEX_USED));
```

Comment:
Could you add these symbols as `static VALUE`s at the top of the file and in the module init? You should see a laundry list sym_foo values initialized around line 1350.